#!/usr/bin/env python3
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Linear PID parameters
gen.add("linear_kp", double_t, 0, "Proportional gain for linear velocity", 1.5, 1.0, 3.0)
gen.add("linear_ki", double_t, 0, "Integral gain for linear velocity", 0.021, 0, 0.3)
gen.add("linear_kd", double_t, 0, "Derivative gain for linear velocity", 0.66, 0.5,0.7)

# Angular PID parameters
gen.add("angular_kp", double_t, 0, "Proportional gain for angular velocity", 2.0, 1.0, 3.0)
gen.add("angular_ki", double_t, 0, "Integral gain for angular velocity", 0.09, 0.01, 0.3)
gen.add("angular_kd", double_t, 0, "Derivative gain for angular velocity", 0.66, 0.5, 0.7)

exit(gen.generate("low_level_pkg", "PID", "PIDConfigConfig"))

